src/transforms/colspec.jl

```diff
@@ -0,0 +1,62 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+# union of types used to select a column
+const ColSelector = Union{Symbol, Integer, AbstractString}
+# union of types used to filter columns
+const ColSpec = Union{Vector{T}, NTuple{N,T}, Regex, Colon} where {N,T<:ColSelector}
+
+# _filter(colspec, allcols)
+# filter table columns using colspec
+#
+# note 1: _filter function always returns a Vector{Symbol},
+# because Vector is more flexible than Tuple.
+#
+# note 2: allcols can be of type Vector or Tuple,
+# in some cases it doesn't matter, but in others it does.
+#
+# note 3: all other methods of the function convert their arguments
+# to call the method below (main method), except the last one.
+
+# _filter(::Vector{Symbol}, allcols) - main method
+# second argument can be a Vector or Tuple
```

Comment:
I believe that this comment section can be omitted, the code already provides this information.